docs/DeveloperGuide.md

```diff
@@ -672,3 +672,75 @@ are equivalent.
 - **Enhancement**: We plan on ensuring that our application collapses multiple spaces
 between individual names such that names like `Alex Yeoh` and `Alex&nbsp;&nbsp;&nbsp;Yeoh`
 can be considered to refer to the same `Name`.
+
+### Enhanced Timestamp Validation Messages
+- **Background**: Many of the transaction-related commands, such as `addTransaction` and `editTransaction`
+may involve a `Timestamp` input. Currently, we validate our `Timestamp` through a combination of regex
+and checking of `DateTimeParseException` from `LocalDateTime.parse(datetime)`. Currently, if
+any validation fails, we show the following error message.
+
+> Date must be in DD/MM/YYYY format and time must be in HH:MM format; date should
+come before time with a single space separating them if both are provided
+
+- **Issue**: This error message is too general, and does not effectively communicate
+to the user why the date is invalid, making it difficult for the user to correct
+the `Timestamp` input in the transaction-related commands.
+
+- **Enhancement**: We plan to make our `Timestamp` validation error messages
+more specific in order to let the user know how they can correct their input. Specifically,
+we plan on covering these 5 cases:
+1. Incorrect Date Format (e.g. `25-06-2023` used dashes instead of slashes)
```

Comment:
If we an detect that they used dashed instead of dots, that implies we can parse the date and accept it

Reply:
Updated